File: examples/governance_static_go/domain/mitigate_incident.go
package domain

import (
	"time"
	"fmt"
)

type MitigateIncident struct {
	IncidentId string `json:"incident_id"`
}

func (c MitigateIncident) CommandName() string { return "MitigateIncident" }

func (c MitigateIncident) Execute(repo IncidentRepository) (*Incident, *MitigatedIncident, error) {
	existing, err := repo.Find(c.IncidentId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("Incident not found: %s", c.IncidentId)
	}
	if existing.Status != "investigating" {
		return nil, nil, fmt.Errorf("cannot MitigateIncident: Incident is in %s state", existing.Status)
	}
	existing.Status = "mitigating"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := MitigatedIncident{
		AggregateID: existing.ID,
		IncidentId: c.IncidentId,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}

File: examples/governance/operations_domain/lib/operations_domain/incident/commands/mitigate_incident.rb
module OperationsDomain
  class Incident
    module Commands
      class MitigateIncident
        include Hecks::Command
        emits "MitigatedIncident"

        attr_reader :incident_id

        def initialize(incident_id: nil)
          @incident_id = incident_id
        end

        def call
          existing = repository.find(incident_id)
          if existing
            unless existing.status == "investigating"
              raise OperationsDomain::Error, "Cannot MitigateIncident: status must be 'investigating', got '#{existing.status}'"
            end
            Incident.new(
              id: existing.id,
              model_id: existing.model_id,
              severity: existing.severity,
              category: existing.category,
              description: existing.description,
              reported_by_id: existing.reported_by_id,
              reported_at: existing.reported_at,
              resolved_at: existing.resolved_at,
              resolution: existing.resolution,
              root_cause: existing.root_cause,
              status: "mitigating"
            )
          else
            raise OperationsDomain::Error, "Incident not found: #{incident_id}"
          end
        end
      end
    end
  end
end
